test/encrypted_schema.test.js

```diff
@@ -0,0 +1,533 @@
+    }
+
+    describe('when a schema is instantiated with a nested encrypted schema', function() {
+      let schema;
+      beforeEach(function() {
+        const encryptedSchema = new Schema({
+          encrypted: {
+            type: String, encrypt: { keyId: KEY_ID, algorithm }
+          }
+        }, { encryptionType: 'csfle' });
+        schema = new Schema({
+          field: encryptedSchema
+        }, { encryptionType: 'csfle' });
+      });
+
+
+      it('then the schema has a nested property that is encrypted', function() {
+        assert.ok(schemaHasEncryptedProperty(schema, ['field', 'encrypted']));
+      });
+    });
+
+    describe('when a schema is instantiated with a nested schema object', function() {
+      let schema;
+      beforeEach(function() {
+        schema = new Schema({
+          field: {
+            encrypted: {
+              type: String, encrypt: { keyId: KEY_ID, algorithm }
+            }
+          }
+        }, { encryptionType: 'csfle' });
+      });
+
+      it('then the schema has a nested property that is encrypted', function() {
+        assert.ok(schemaHasEncryptedProperty(schema, ['field', 'encrypted']));
+      });
+    });
+
+    describe('when a schema is instantiated as an Array', function() {
+      let schema;
+      beforeEach(function() {
+        schema = new Schema({
+          encrypted: {
+            type: [Number],
+            encrypt: { keyId: KEY_ID, algorithm }
+          }
+        }, { encryptionType: 'csfle' });
+      });
+
+      it('then the schema has a nested property that is encrypted', function() {
+        assert.ok(schemaHasEncryptedProperty(schema, 'encrypted'));
+      });
+    });
+
+  });
+
+  describe('invalid schema types for encrypted schemas', function() {
+    describe('When a schema is instantiated with an encrypted field of type Number', function() {
+      it('Then an error is thrown', function() {
+        assert.throws(() => {
+          new Schema({
+            field: {
+              type: Number, encrypt: { keyId: KEY_ID, algorithm }
+            }
+          }, { encryptionType: 'csfle' });
+        }, /unable to determine bson type for field `field`/);
+      });
+    });
+
+    describe('When a schema is instantiated with an encrypted field of type Mixed', function() {
+      it('Then an error is thrown', function() {
+        assert.throws(() => {
+          new Schema({
+            field: {
+              type: Schema.Types.Mixed, encrypt: { keyId: KEY_ID, algorithm }
+            }
+          }, { encryptionType: 'csfle' });
+        }, /unable to determine bson type for field `field`/);
+      });
+    });
+
+    describe('When a schema is instantiated with a custom schema type plugin', function() {
+      class Int8 extends mongoose.SchemaType {
+        constructor(key, options) {
+          super(key, options, 'Int8');
+        }
+      }
+
+      // Don't forget to add `Int8` to the type registry
+      mongoose.Schema.Types.Int8 = Int8;
```

Comment:
Should this be done in before/after? does it matter that there's now an Int8 type globally?